pkg/apiserver/filters/impersonation.go

```diff
@@ -123,6 +123,15 @@ func WithImpersonation(handler http.Handler, requestContextMapper api.RequestCon
 		oldUser, _ := api.UserFrom(ctx)
 		httplog.LogOf(req, w).Addf("%v is acting as %v", oldUser, newUser)
 
+		// clear all the impersonation headers from the request
+		req.Header.Del(authenticationapi.ImpersonateUserHeader)
```

Comment:
I understand and totally agree with your first part, but am sorry I didnt understand the second part.

Isnt front proxy just a user of the generic "impersonation framework"?